tests/conftest.py

```diff
@@ -160,3 +174,96 @@ def metadata_model(helpers, data_model_labels):
     )
 
     return metadata_model
+
+
+@pytest.fixture(scope="function")
+def schedule_for_cleanup(
+    request, synapse_store: SynapseStorage
+) -> Callable[[CleanupItem], None]:
+    """Returns a closure that takes an item that should be scheduled for cleanup."""
+
+    items: Set[CleanupItem] = set()
+
+    def _append_cleanup(item: CleanupItem):
+        print(f"Added {item} to cleanup list")
+        items.add(item)
+
+    def cleanup_scheduled_items() -> None:
+        for item in items:
+            print(f"Cleaning up {item}")
+            try:
+                if item.action == CleanupAction.DELETE:
+                    if item.synapse_id:
+                        synapse_store.syn.delete(obj=item.synapse_id)
+                    elif item.name and item.parent_id:
+                        synapse_id = synapse_store.syn.findEntityId(
+                            name=item.name, parent=item.parent_id
+                        )
+                        if synapse_id:
```

Comment:
Testing this out and it looks like `findEntityId` silently returns nothing if a given `name` and `parent` don't match up with what exists on Synapse. Should we add a `logger.error("Synapse ID not found")`

<img width="513" alt="image" src="https://github.com/user-attachments/assets/46127fdf-c160-457c-9d8e-3c24485806cf">

Reply:
It can also mean that the content was already deleted and is in a trash can. Either way, the spirit of what needs to be accomplished is completed if it cannot be found as we want it to be deleted.

For other actions, if they're added, we might want to have that in place.